src/test/java/uk/gov/justice/hmpps/prison/service/OffenderEmploymentServiceTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.justice.hmpps.prison.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.Occupation;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderEmployment;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderEmployment.EmploymentPostType;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderEmployment.PK;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderEmployment.PayPeriodType;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderEmployment.ScheduleType;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderEmploymentAddress;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderEmploymentRepository;
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderEmploymentTransformer;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class OffenderEmploymentServiceTest {
+
+    @Mock
+    private OffenderEmploymentRepository repository;
+
+    private final OffenderEmploymentTransformer transformer = new OffenderEmploymentTransformer();
+
+    private OffenderEmploymentService service;
+
+
+    @BeforeEach
+    void setup() {
+        service = new OffenderEmploymentService(repository, transformer);
+    }
+
+
+    @Test
+    public void getOffenderEmployments() {
+        var nomisId = "abc";
+        var pageRequest = PageRequest.of(0, 10);
+        var offenderEmployments = generateMockData();
+
+        when(repository.findAllByNomisId(nomisId, pageRequest)).thenReturn(offenderEmployments);
+
+        var actual = service.getOffenderEmployments(nomisId, pageRequest).getContent();
+        var expected = offenderEmployments.getContent().stream().map(transformer::convert).collect(Collectors.toList());
```

Comment:
I'm not really sure that you should generate expectations by calling the transformer - you might as well just mock the transformer and assert it was called.  As a rule of thumb test expectations shouldn't be generated from production code.